source/slang/glsl.meta.slang

```diff
@@ -2825,21 +2825,1988 @@ public uint rayQueryGetIntersectionTypeEXT(rayQueryEXT q, bool committed)
     return 0;
 }
 
+// TODO: implementation of built-in variables; proper tests; these are stubs
+// likley related to the following issue since GLSL adds new 
+// 'system' variables: https://github.com/shader-slang/slang/issues/411
+
+#define glsl_caps_shader_subgroup(param1) __glsl_extension(param1) [require(glsl)] 
+#define spirv_caps_shader_subgroup(param1) __spirv_version(param1) [require(spirv)] 
+
+//macros used to make implementation and modification reasonable under regex
+#define macroAnyVecGenericBuiltIn __generic<T : __BuiltinType, let N : int>
+#define macroAnyVecGenericLogical __generic<T : __BuiltinLogicalType, let N : int>
+#define macroAnyVecGenericArithmetic __generic<T : __BuiltinArithmeticType, let N : int>
+#define macroAnyVec vector<T, N>
+
+__generic<T : __BuiltinType>
+void shader_subgroup_preamble() {
+    // checks needed for shader_subgroup functions; __requireGLSLExtension does not work, the call fails to find the OP
```

Comment:
I am not sure what this comment means.
It says __requireGLSLExtension does not work, but you are using it at 4 lines below.
And shouldn't you be using "__glsl_extension", instead?

Reply:
as done in hlsl.meta.slang (`WaveMaskPrefixSum`), 
``` if (__isHalf<T>()) __requireGLSLExtension("GL_EXT_shader_subgroup_extended_types_float16"); ``` 
is used to enable an extension which is required based on type used with an ext.

despite this, __requireGLSLExtension does not work with glsl targeted files (glsl.meta.slang)

Reply:
Again, what do you mean by "not work".
Does it causes a compile error?
Or is the requirement applied to all types?
Or is the line doing nothing?

Reply:
__requireGLSLExtension is doing nothing, but it should be doing something (adding the extension to the final compiled glsl code)